[Sien Soon] iP

ssyap98 opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/505).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [X] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

Commit:
Add Gradle support

ssyap98 committed:
Added Level-1: Greet, Echo, Exit functions

ssyap98 committed:
Added Level-2: Add, list functions

ssyap98 committed:
Added Level-3: Mark as Done function

ssyap98 committed:
Added Level-4: ToDos, Events, Deadlines

ssyap98 committed:
Added A-TextUiTesting: Automated Text UI Testing

ssyap98 committed:
Added Level-5: Handle Errors
Edited EXPECTED.TXT for TextUiTesting

ssyap98 committed:
Added Level-6: Delete tasks from list

ssyap98 committed:
Add Level-7: Save

ssyap98 committed:
Add Level-8: Dates and Times

ssyap98 committed:
Merge branch 'branch-Level-8'

BlopApple commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Maybe you could name it as "deadlineTime" instead of "by"?

BlopApple commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,235 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        duke.greeting();
+        duke.run();
+        duke.goodBye();
```
Maybe you could name the methods as verbs like greet() and exit() instead?

BlopApple commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,235 @@
+                // It is then a special character in regular expressions which means "use the next character literally, don't interpret its special meaning"
+                String taskType = taskInArray[0];
+                switch (taskType) {
+                    case "T": {
+                        Task task = new ToDo(taskInArray[2]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "D": {
+                        Task task = new Deadline(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "E": {
+                        Task task = new Event(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
```
Since in each case you are marking the task as done when loading them, maybe u can abstract the process out?

BlopApple reviewed (commented):
Overall your code looks fine, although could probably use a bit more abstraction for future implementations.

TYKCodes commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,235 @@
+
 public class Duke {
+
+    private final String horizontalLine = "-------------------------";
```
You may want the name of your constant variables like this one to match the coding standard. Perhaps something like:
```suggestion
    private final String HORIZONTAL_LINE = "-------------------------";
```

TYKCodes commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,235 @@
+                String[] taskInArray = taskInFile.split(" \\| "); //The \\ is really equivalent to a single \ (the first \ is required as a Java escape sequence in string literals).
+                // It is then a special character in regular expressions which means "use the next character literally, don't interpret its special meaning"
+                String taskType = taskInArray[0];
+                switch (taskType) {
+                    case "T": {
+                        Task task = new ToDo(taskInArray[2]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "D": {
+                        Task task = new Deadline(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
+
+                    case "E": {
+                        Task task = new Event(taskInArray[2], taskInArray[3]);
+                        taskList.add(task);
+                        if (taskInArray[1].equals("1")) {
+                            task.markAsDone();
+                        }
+                        break;
+                    }
```
The indentation for each case statement should be aligned to the switch

TYKCodes reviewed (commented):
Overall, the code looks clean and understandable. Just a few touching up of the formatting and it should be all good

ssyap98 committed:
Add A-MoreOOP

ssyap98 committed:
Organise all classes into one package

ssyap98 committed:
Organise all classes into 1 package

ssyap98 committed:
Move all classes into the duke folder

ssyap98 committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

ssyap98 committed:
Save Event, Deadline and ToDo tasks to file when created

ssyap98 committed:
Add JUnit Tests for Event and ToDo class

ssyap98 committed:
Change mainClassName and archiveBaseName in build.gradle

ssyap98 committed:
Add JavaDoc comments for Duke.java

ssyap98 committed:
Use plural form to represent a list of tasks

ssyap98 committed:
Add ability to find a task by searching for a keyword

ssyap98 committed:
Merge branch 'branch-A-JavaDoc'

ssyap98 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/TaskList.java

ssyap98 committed:
Change list to tasks to follow coding standard

ssyap98 committed:
Create GUI for Duke

ssyap98 committed:
Merge branch 'branch-Level-10'

ssyap98 committed:
Adds date and time to Event class

To allow Duke to understand dates and times instead of string for event schedule.

Store dates and times in format (yyyy-mm-ddThh:mm:ss) and prints in a different format (MMM dd yyyy hh:mm:ss).

ssyap98 committed:
Adds flip() method in DialogBox

Flips the dialog box such that the image on the left corresponds to Duke's output while the image on the right corresponds to the user's input.

ssyap98 committed:
Adds assertion to document important assumptions

ssyap98 committed:
Fix code quality in Storage class

Avoid arrowhead style code

ssyap98 committed:
Merge pull request #2 from ssyap98/branch-A-Assertions

Adds assertion to document important assumptions

ssyap98 committed:
Merge branch 'master' into branch-A-CodeQuality

ssyap98 committed:
Merge pull request #3 from ssyap98/branch-A-CodeQuality

Fix code quality in Storage class

ssyap98 committed:
Fix minor bug

ssyap98 committed:
Add case "bye" to parser which was accidentally removed earlier

ssyap98 committed:
Adds Category C extension: C-Help

ssyap98 committed:
Merge branch 'branch-C-Help'

ssyap98 committed:
Change some phrases used by Duke when responding to commands

ssyap98 committed:
Adds User Guide and Ui screenshot

ssyap98 committed:
Update README.md

ssyap98 committed:
Create MainWindow.fxml and DialogBox.fxml in src/main/resources/view

ssyap98 committed:
Use FXML for Duke

ssyap98 committed:
Replace image name

ssyap98 committed:
Add welcome/greeting message upon running Duke

ssyap98 committed:
Fix "bye" command

Exits Duke when "bye" is entered by user

ssyap98 committed:
Fix error message when unknown command is entered by user

ssyap98 committed:
Add method to show error message when invalid todo, deadline, event, mark, unmark, delete and find is performed

ssyap98 committed:
Set purple color background for dialog box

ssyap98 committed:
Change dialogbox font size and color

ssyap98 committed:
Change mainwindow background color to black

ssyap98 committed:
Set dialogbox's minimum height

ssyap98 committed:
Add header comments in Deadline.java

ssyap98 committed:
Remove empty space

ssyap98 committed:
Add header comments in DialogBox.java

ssyap98 committed:
Update header comment in Deadline.java

ssyap98 committed:
Add header comments in Duke.java

ssyap98 committed:
Add header comments in DukeException.java

ssyap98 committed:
Add header comments in Event.java

ssyap98 committed:
Add header comments in Parser.java

ssyap98 committed:
Add header comments in Storage.java

ssyap98 committed:
Add header comments in Task.java

ssyap98 committed:
Add header comments in TaskList.java

ssyap98 committed:
Add header comments in ToDo.java

ssyap98 committed:
Add header comments in Ui.java
